### PR TITLE
[Merged by Bors] - Simplify error handling after engines fallback removal

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed
Part of #3118, continuation of #3257

## Proposed Changes
- the [ `first_success_without_retry` ](https://github.com/sigp/lighthouse/blob/9c429d0764ed91cf56efb8a47a35a556b54a86a4/beacon_node/execution_layer/src/engines.rs#L348-L351) function returns a single error.
- the [`first_success`](https://github.com/sigp/lighthouse/blob/9c429d0764ed91cf56efb8a47a35a556b54a86a4/beacon_node/execution_layer/src/engines.rs#L324) function returns a single error.
- [ `EngineErrors` ](https://github.com/sigp/lighthouse/blob/9c429d0764ed91cf56efb8a47a35a556b54a86a4/beacon_node/execution_layer/src/lib.rs#L69) carries a single error.
- [`EngineError`](https://github.com/sigp/lighthouse/blob/9c429d0764ed91cf56efb8a47a35a556b54a86a4/beacon_node/execution_layer/src/engines.rs#L173-L177) now does not need to carry an Id
- [`process_multiple_payload_statuses`](https://github.com/sigp/lighthouse/blob/9c429d0764ed91cf56efb8a47a35a556b54a86a4/beacon_node/execution_layer/src/payload_status.rs#L46-L50) now doesn't need to receive an iterator of statuses and weight in different errors

## Additional Info
This is built on top of #3294

### CODE REVIEW

**Comment by divagant-martian** [comment]
test failure is fixed by https://github.com/sigp/lighthouse/pull/3303

File: beacon_node/execution_layer/src/engines.rs [hunk 1]
```diff
@@ -261,45 +255,35 @@ impl Engines {
...
+    pub async fn first_success<'a, F, G, H>(&'a self, func: F) -> Result<H, EngineError>
     where
         F: Fn(&'a Engine<EngineApi>) -> G + Copy,
         G: Future<Output = Result<H, EngineApiError>>,
     {
         match self.first_success_without_retry(func).await {
             Ok(result) => Ok(result),
-            Err(mut first_errors) => {
-                // Try to recover some nodes.
+            Err(_e) => {
```
Comment (paulhauner) [inline beacon_node/execution_layer/src/engines.rs]: Perhaps we could log this error to assist with debugging?

**Review by paulhauner — APPROVED** [review]
Nice and succinct! I'm happy with this, I just had one suggestion about error logging.

File: beacon_node/execution_layer/src/engines.rs (hunk 1 above)
Comment (divagant-martian) [reply beacon_node/execution_layer/src/engines.rs]: let me know if you prefer a different msg https://github.com/sigp/lighthouse/pull/3283/commits/7150a89de2235e02043288b700e95e047460423f

**Review by paulhauner — APPROVED** [review]
Looks good! Merge at will :relaxed:

**Pull request closed without merging**